packages/web/src/services/FileDownloadServiceWeb.ts

```diff
@@ -1,76 +1,119 @@
-            return this.downloadS3Directory(fileInfo);
+            const downloadResult = await this.handleZarrFile(fileInfo);
+            if (downloadResult) {
+                return downloadResult;
+            }
         }
 
-        const data = fileInfo.data || fileInfo.path;
-        let downloadUrl: string;
-        if (data instanceof Uint8Array) {
-            downloadUrl = URL.createObjectURL(new Blob([data]));
-        } else if (data instanceof Blob) {
-            downloadUrl = URL.createObjectURL(data);
-        } else {
-            downloadUrl = data;
+        return this.downloadFile(fileInfo);
+    }
+
+    private async handleZarrFile(fileInfo: FileInfo): Promise<DownloadResult | null> {
+        if (this.isS3Url(fileInfo.path)) {
+            return await this.downloadS3Directory(fileInfo);
         }
 
-        try {
-            const a = document.createElement("a");
-            a.href = downloadUrl;
-            a.download = fileInfo.name;
-            a.target = "_blank";
-            a.click();
-            a.remove();
+        const localDownloadResult = this.isLocalPath(fileInfo.path, fileInfo);
+        return localDownloadResult; // This will be either a DownloadResult or null
+    }
+
+    private isLocalPath(filePath: string, fileInfo: FileInfo): DownloadResult | null {
+        const uriPattern = /^(https?|ftp):\/\/|^[a-zA-Z]:\\/;
+        const isLocal = filePath.startsWith("file://") || !uriPattern.test(filePath);
+
+        if (isLocal) {
+            const directoryPath = fileInfo.path;
+
+            const message = `The directory containing the Zarr file is located at: ${directoryPath}.
+Due to security restrictions, the web browser cannot open this location directly. 
+Please navigate to this directory manually, or upload files to a remote address such as S3.`;
+
+            alert(message);
+            console.log(`Local directory path: ${directoryPath}`);
+
             return {
                 downloadRequestId: fileInfo.id,
-                resolution: DownloadResolution.SUCCESS,
+                resolution: DownloadResolution.CANCELLED,
             };
-        } catch (err) {
-            console.error(`Failed to download file: ${err}`);
-            throw err;
-        } finally {
-            URL.revokeObjectURL(downloadUrl);
         }
+
+        return null;
     }
 
     private async downloadS3Directory(fileInfo: FileInfo): Promise<DownloadResult> {
-        const { bucket, key, region } = this.parseS3Url(fileInfo.path);
-        const keys = await this.listS3Objects(bucket, key, region);
+        const { hostname, key } = this.parseS3Url(fileInfo.path);
+        const keys = await this.listS3Objects(hostname, key);
 
         if (keys.length === 0) {
             throw new Error("No files found in the specified S3 directory.");
         }
 
-        const zip = new JSZip();
+        const fileStream = StreamSaver.createWriteStream(`${fileInfo.name}.zip`);
```

Comment:
Confirmed that the zip is just all the files stitched together. The original method I had I tested and extracted the output. With this new system, I have just been looking at the exported zip file size since the feature isnt finished. Trying to unzip it now does not work :(. Looking into other options